src/smart-components/role/role-permissions-table-helpers.js

```diff
@@ -18,7 +18,15 @@ export const createRows = (showResDefinitions, uuid) => (data, opened, selectedR
             ? [
                 permission.includes('cost-management') && resourceDefinitions.length > 0 ? (
                   <Fragment key="resource-definitions">
-                    <Link to={`/roles/detail/${uuid}/permission/${permission}`}>{resourceDefinitions.length}</Link>
+                    <Link to={`/roles/detail/${uuid}/permission/${permission}`}>
+                      {resourceDefinitions.reduce((acc, curr) => {
+                        if (curr.attributeFilter?.value) {
+                          return acc + (Array.isArray(curr.attributeFilter.value) ? curr.attributeFilter.value.length : 1);
+                        } else {
+                          return acc;
+                        }
+                      }, 0)}
```

Comment:
I don't really like `return`s on random lines, this might be slightly better
```suggestion
                      {resourceDefinitions.reduce((acc, { attributeFilter: { value } } = { attributeFilter: {} }) => (
                        acc + (value ? (Array.isArray(value) ? value.length : 1) : 0)
                      ), 0)}
```

Reply:
replaced the whole reduce, it looks much better :)